department-of-veterans-affairs/caseflow
# Add Travel Board ("T") Docket Type

Resolves [CASEFLOW-1820](https://vajira.max.gov/browse/CASEFLOW-1820)

### Description
Add the Travel Board Docket type, but force it to not have "Travel" hearings. That's why there's a `request_type` override in hearings.rb

### Contents
This PR ended up having a few different pieces. Here's one way to think about them:
- Changes to models/rails (hearing.rb, hearing_day.rb, factory, and specs)
- Changes so that 9am/1pm are the available half day times on the frontend. Note that the AC says "8:30am/1pm, but that's not actually what we use for Central, we use 9am/1pm, so I went with that.
    - AddHearingDay and DocketStartTimes go together and do the same job as
    - ScheduleVeteranForm and modalForms/HearingTime (beware, there's another HearingTime component)
    - I had to almost fully rewrite DocketStartTimes to allow it to work correctly.

### Testing Plan
You should be able to test most of this PR by manually Adding hearing days, but this PR contains seed data changes, to test those changes you'll need to run `make reset` and wait until that finishes to fully test this PR.

Steps:
- From [hearing schedule page](http://localhost:3000/hearings/schedule), click the "Add Hearing Day" button.
  - Create a Travel Hearing Day using that form for some future date
- Back on the hearing schedule page, check that your Travel day shows in the list and that the filter works. If you ran `make reset` you'll see a bunch of Travel days.
- Schedule a veteran for a virtual hearing on your travel day.
- Schedule a veteran for a video hearing on your travel day.
  - Note that you can't schedule a veteran for a "Travel" hearing this is an AC.
  - Note that the times are 9am/1pm, this is an AC.
- Go to the daily docket for the hearings you just scheduled, make sure it looks correct
  - These hearings should be either "Video" or "Virtual", there's no way to have a "Travel" hearing yet.
  - Record the URL of the daily docket so you have the ID: http://localhost:3000/hearings/schedule/docket/396
- Examine the HearingDay in the console
  - `hd = HearingDay.find(396) # Find the HearingDay using the id on the end of that URL`
  - `hd.total_slots # Should be 10`
  - `hd.slot_length_minutes # Should be 60`
  - `hd.begins_at # Should be nil`, gets set on the front end in `TimeSlot.jsx`
